### PR TITLE
 Add the rollback test in the data pipeline - fix #275 

### LINKED ISSUE #275: Bug snapshot test should not be a CI test, but a test run after training
It is more of an integration test.

We should run it on the DB fixtures on CI, and run it on the entire DB only after training.

### PR DESCRIPTION
**Pull request by Lothiraldan**
Fixes #275.

### CODE REVIEW

File: .taskcluster.yml [hunk 1]
```diff
@@ -113,7 +113,8 @@ tasks:
                cd bugbug &&
                git checkout ${head_rev} &&
                pip install -r requirements.txt &&
-               python -c 'from bugbug import bugzilla, db; db.download()' &&
+               mkdir data/ &&
+               cp tests/fixtures/bugs.json data/bugs.json &&
                python -m bugbug.bug_snapshot"
```
Comment (marco-c) [inline .taskcluster.yml]: We could turn it into a pytest, that'd be nicer

Comment (marco-c) [reply .taskcluster.yml]: So we can completely remove this task

File: infra/data-pipeline.yml [hunk 2]
```diff
@@ -164,6 +164,37 @@ tasks:
       owner: release-mgmt-analysis@mozilla.com
       source: https://github.com/mozilla/bugbug/raw/master/data-pipeline.yml
 
+  - ID: rollback-test-task
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '49 hours'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc
+    dependencies:
```
Comment (marco-c) [inline infra/data-pipeline.yml]: The rollback test should be able to run right after the download bugzilla task.

File: tests/test_bug_snapshot.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from bugbug.bug_snapshot import rollback
+
+
+def test_bug_snapshot(get_fixture_path):
+    mock_bugs_db_path = get_fixture_path("bugs.json")
+
+    with open(mock_bugs_db_path, "r") as mock_bug_db:
+
+        for i, line in enumerate(mock_bug_db):
+            bug = json.loads(line)
```
Comment (marco-c) [inline tests/test_bug_snapshot.py]: I think you could replace all this (from line 13 to 18) with `for bug in bugzilla.get_bugs():` (it should work as the mock bug DB is stored in data/)

File: tests/test_bug_snapshot.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+            print(bug["id"])
+            print(i)
```
Comment (marco-c) [inline tests/test_bug_snapshot.py]: Nit: remove the printing

File: infra/data-pipeline.yml [hunk 5]
```diff
@@ -56,6 +56,34 @@ tasks:
...
+  - ID: rollback-test-task
+    created: {$fromNow: ''}
+    deadline: {$fromNow: '25 hours'}
+    provisionerId: aws-provisioner-v1
+    workerType: releng-svc
+    dependencies:
+      - bugs-retrieval
+    payload:
+      maxRunTime: 3600
+      image: python
```
Comment (marco-c) [inline infra/data-pipeline.yml]: Probably better to use the bugbug base image, so we don't have to install the requirements (and we don't have to define yet another image)

File: infra/data-pipeline.yml [hunk 6]
```diff
@@ -56,6 +56,34 @@ tasks:
...
+      command:
+        - "/bin/bash"
+        - "-lcx"
+        - "git clone ${repository} &&
+           cd bugbug &&
+           git checkout ${head_rev} &&
```
Comment (marco-c) [inline infra/data-pipeline.yml]: Related to the prev comment, head_rev is undefined here, so it wouldn't know what to checkout to. If we use the base image, this should be fixed automatically

File: tests/test_bug_snapshot.py (hunk 4 above)
Comment (Lothiraldan) [reply tests/test_bug_snapshot.py]: As we are in a loop, I think the prints will be helpful to quickly identify which bug is triggering an error. It's also helpful to check that the mock DB is correctly read by running `pytest -s`

Comment (marco-c) [reply tests/test_bug_snapshot.py]: OK, sounds good!

**Pull request merged**